source/class/qx/ui/form/DateField.js

```diff
@@ -602,7 +602,22 @@ qx.Class.define("qx.ui.form.DateField",
 
       field.getFocusElement().focus();
       field.selectAllText();
-    }
+    },
+
+    // overridden
+    setAriaLabel: function(label) {
+      this.getChildControl("textfield").setAriaLabel(label);
+    },
+
+    // overridden
+    addAriaLabelledBy: function(...labelWidgets) {
```

Comment:
this will require some `arguments` and `apply` or `call` magic when backporting, but it would certainly be possible.